### PR TITLE
Baremetal:  update networking documentation - API LB

### PR DESCRIPTION
**Pull request by yboaron**
Describe API load balancing flow for internal and external clients.

### CODE REVIEW

File: docs/design/baremetal/networking-infrastructure.md [hunk 1]
```diff
@@ -69,6 +70,11 @@ https://github.com/openshift/machine-config-operator/pull/795
 See [here](FIXME: link to a README in MCO) for more detailed information about
 the `haproxy` configuration.
 
+For clients inside the cluster, the Kubernetes apiserver is accessible via the
```
Comment (abhinavdahiya) [inline docs/design/baremetal/networking-infrastructure.md]: This information is technically not baremetal specific and it's general kubernetes.. I don't think this is required.

Comment (yboaron) [reply docs/design/baremetal/networking-infrastructure.md]: @abhinavdahiy - that's correct, will delete this part

**Pull request merged**